sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/policy/UrlSanitizer.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.http.policy;
+
+import com.azure.core.implementation.ImplUtils;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.UrlBuilder;
+
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.azure.core.implementation.logging.LoggingKeys.REDACTED_PLACEHOLDER;
+
+public final class UrlSanitizer {
+    static final List<String> DEFAULT_QUERY_PARAMS_ALLOWLIST = Collections.singletonList("api-version");
+    private final Set<String> allowedQueryParamNames;
+
+    public UrlSanitizer(Collection<String> allowedQueryParamNames) {
+        Collection<String> allowedParams
+            = allowedQueryParamNames == null ? DEFAULT_QUERY_PARAMS_ALLOWLIST : allowedQueryParamNames;
```

Comment:
I think `api-version` should always be included in the allowed query params list. So, if the custom list is provided that doesn't include api-version, we should add it.